spread.yaml

```diff
@@ -263,6 +268,11 @@ suites:
                 prepare_classic
             fi
         restore: |
+            if [ "$REMOTE_STORE" = staging ]; then
+                . $TESTSLIB/store.sh
+                teardown_staging_store
+            fi
+
             $TESTSLIB/reset.sh
```

Comment:
Indeed, we handle a bit different prepare and restore (prepare sources and calls functions, restore just calls the executable reset.sh file and the functions are called from inside), I've just moved the store teardown to reset.sh, plz have a look.